src/Engine/ProtoCore/DSASM/Heap.cs

```diff
@@ -624,40 +624,45 @@ private bool TryFindFreeIndex(out int index)
             }
         }
 
+        //cache ClassIndex and ProcdureNode for repeated calls of the same type object.
+        private int previousClassIndex;
+        private ProcedureNode pn;
+
         private void GCDisposeObject(StackValue svPtr, Executive exe)
         {
             int classIndex = svPtr.metaData.type;
-            ClassNode cn = exe.exe.classTable.ClassNodes[classIndex];
 
-            ProcedureNode pn = cn.GetDisposeMethod();
-            while (pn == null)
+            if (this.pn == null || classIndex != previousClassIndex)
             {
-                if (cn.Base != Constants.kInvalidIndex)
-                {
-                    classIndex = cn.Base;
-                    cn = exe.exe.classTable.ClassNodes[classIndex];
-                    pn = cn.GetDisposeMethod();
-                }
-                else
+                previousClassIndex = classIndex;
+                ClassNode cn = exe.exe.classTable.ClassNodes[classIndex];
+
+                pn = cn.GetDisposeMethod();
+                while (pn == null)
                 {
-                    break;
+                    if (cn.Base != Constants.kInvalidIndex)
+                    {
+                        classIndex = cn.Base;
+                        cn = exe.exe.classTable.ClassNodes[classIndex];
+                        pn = cn.GetDisposeMethod();
+                    }
+                    else
+                    {
+                        break;
+                    }
                 }
             }
 
             if (pn != null)
             {
-                // TODO Jun/Jiong: Use build pointer utilities 
-                exe.rmem.Push(StackValue.BuildArrayDimension(0));
-                exe.rmem.Push(StackValue.BuildPointer(svPtr.Pointer, svPtr.metaData));
-                exe.rmem.Push(StackValue.BuildInt(1));
-                
```

Comment:
Why have these lines been removed?

Reply:
This was not necessary with the new route and would be a memory leak.  Also FYI in Dynamo today it does seem to have a memory leak regardless.  We always add 3 items but only ever Pop two.